server/portal/apps/workspace/api/views.py

```diff
@@ -151,6 +135,53 @@ def get(self, request, *args, **kwargs):
             encoder=BaseTapisResultSerializer
         )
 
+    def select(self, client, request):
+        job_uuid = request.GET.get('job_uuid')
+        data = client.jobs.getJob(jobUuid=job_uuid)
+
+        return data
+
+    def listing(self, client, request):
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        data = client.jobs.getJobSearchList(
+            limit=limit,
+            startAfter=offset,
+            orderBy='lastUpdated(desc),name(asc)',
+            _tapis_query_parameters={'tags.contains': portal_name}
+        )
+
+        return data
+
+    def search(self, client, request):
+
+        query_string = request.GET.get('query_string')
+
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        sql_queries = [
+            f"(tags IN ('{portal_name}')) AND",
+            f"(name like '%{query_string}%') OR",
+            f"(archiveSystemDir like '%{query_string}%') OR",
+            f"(appId like '%{query_string}%') OR",
+            f"(archiveSystemId like '%{query_string}%')",
+        ]
+
```

Comment:
should we look into making the query return case-insensitive results?  

maybe a follow on ticket/PR?